app/DoctrineMigrations/Version20190129120000.php

```diff
@@ -0,0 +1,152 @@
+        [
+            'name' => 'share_twitter',
+            'value' => '1',
+            'section' => 'entry',
+        ],
+        [
+            'name' => 'show_printlink',
+            'value' => '1',
+            'section' => 'entry',
+        ],
+        [
+            'name' => 'export_epub',
+            'value' => '1',
+            'section' => 'export',
+        ],
+        [
+            'name' => 'export_mobi',
+            'value' => '1',
+            'section' => 'export',
+        ],
+        [
+            'name' => 'export_pdf',
+            'value' => '1',
+            'section' => 'export',
+        ],
+        [
+            'name' => 'export_csv',
+            'value' => '1',
+            'section' => 'export',
+        ],
+        [
+            'name' => 'export_json',
+            'value' => '1',
+            'section' => 'export',
+        ],
+        [
+            'name' => 'export_txt',
+            'value' => '1',
+            'section' => 'export',
+        ],
+        [
+            'name' => 'export_xml',
+            'value' => '1',
+            'section' => 'export',
+        ],
+        [
+            'name' => 'piwik_enabled',
+            'value' => '0',
+            'section' => 'analytics',
+        ],
+        [
+            'name' => 'piwik_host',
+            'value' => 'v2.wallabag.org',
+            'section' => 'analytics',
+        ],
+        [
+            'name' => 'piwik_site_id',
+            'value' => '1',
+            'section' => 'analytics',
+        ],
+        [
+            'name' => 'demo_mode_enabled',
+            'value' => '0',
+            'section' => 'misc',
+        ],
+        [
+            'name' => 'demo_mode_username',
+            'value' => 'wallabag',
+            'section' => 'misc',
+        ],
+        [
+            'name' => 'wallabag_support_url',
+            'value' => 'https://www.wallabag.org/pages/support.html',
+            'section' => 'misc',
+        ],
+    ];
+
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        foreach ($this->settings as $setting) {
+            $settingEnabled = $this->container
+                ->get('doctrine.orm.default_entity_manager')
+                ->getConnection()
+                ->fetchArray('SELECT * FROM ' . $this->getTable('craue_config_setting') . " WHERE name = '" . $setting['name'] . "'");
+
+            if (false !== $settingEnabled) {
+                continue;
+            }
+
+            $this->addSql('INSERT INTO ' . $this->getTable('craue_config_setting') . " (name, value, section) VALUES ('" . $setting['name'] . "', '" . $setting['value'] . "', '" . $setting['section'] . "');");
+        }
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+        foreach ($this->settings as $setting) {
+            $this->addSql('
+                DELETE FROM ' . $this->getTable('craue_config_setting') . "
```

Comment:
I'm quite uncomfortable with this one. A down of this migration may lead to a deletion of legitimate settings. Any thoughts @j0k3r?

Reply:
Well we always provide a _down_ migration.
But maybe in that case, we shouldn't because these settings are required.